File: backend/app/models/prediction.py
from datetime import datetime
from sqlalchemy import String, Float, DateTime, func
from sqlalchemy.orm import Mapped, mapped_column
from app.database.base import Base


class Prediction(Base):
    __tablename__ = "predictions"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    symbol: Mapped[str] = mapped_column(String(50), nullable=False)
    category: Mapped[str] = mapped_column(String(50), nullable=False)  # EQUITY, INTRADAY, OPTIONS
    horizon: Mapped[str] = mapped_column(String(20), nullable=False)    # 1D, 5D, 20D
    direction: Mapped[str] = mapped_column(String(10), nullable=False)  # UP, DOWN, SIDEWAYS
    confidence: Mapped[float] = mapped_column(Float, nullable=False)
    regime: Mapped[str] = mapped_column(String(50), nullable=False)
    model_version: Mapped[str] = mapped_column(String(50), nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)

File: frontend/src/types/prediction.ts
export interface StockPrediction {
  id: string;
  symbol: string;
  name: string;
  category: 'EQUITY' | 'INTRADAY' | 'OPTIONS';
  direction: 'UP' | 'DOWN' | 'SIDEWAYS';
  targetPrice?: number;
  currentPrice: number;
  confidence: number;
  horizon: '1D' | '5D' | '20D';
  regime: string;
  modelVersion: string;
  shapTopFeatures?: { feature: string; impact: number }[];
}
